app/scripts/controllers/transactions/index.js

```diff
@@ -739,6 +791,10 @@ export default class TransactionController extends EventEmitter {
       sendFlowHistory,
     });
 
+    if (actionId) {
+      txMeta.actionId = actionId;
+    }
```

Comment:
Should we not do this in the existing if block that checks for actionId?

Reply:
We add this to only fresh transaction.

Reply:
I think what Brad was saying it the check on :796 and :767 is the same thing...why not do :797 within the :767 block.

Reply:
`txMeta` gets created on line :789, after :767.  This block isn't needed if the :767 block finds an existing transaction, in which case, it will end the function then and not get to these lines

Reply:
@brad-decker @jpuri @segun 

`txMeta` gets created on line :789, after :767. This block isn't needed if the :767 block finds an existing transaction, in which case, it will end the function then and not get to these lines

I think we can delete the `// Add actionId to txMeta to check if same actionId is seen again` comment because there's an explanation of what `actionId` is used for in the comment for the function. This comment is also missing context that might cause confusion